pages/docs/sdk/python/otel_based_sdk_v3.mdx

```diff
@@ -0,0 +1,924 @@
+
+3.  **Creating Traces and Observations**:
+
+    - **v2**: `langfuse.trace()`, `langfuse.span()`, `langfuse.generation()`. These were distinct objects.
+    - **v3**:
+      - A trace is implicitly created by the first (root) span or generation. There is no direct `langfuse.trace()` method.
+      - Use `langfuse.start_as_current_span()`, `langfuse.start_as_current_generation()` (context managers) or `langfuse.start_span()`, `langfuse.start_generation()` (manual `end()`).
+      - The `name` parameter is now **required** for all spans and generations and cannot be updated later (only via attributes that must be parsed server-side).
+      - The Langfuse Event observation type has been dropped.
+
+4.  **Ending Observations**:
+
+    - **v2**: Some objects might have auto-ended or relied on `update()` with an optional `end_time`.
+    - **v3**: All spans and generations **must be explicitly ended** by calling their `.end()` method, or by using them as context managers (`with ... as ...:`), which handles ending automatically. Not ending spans will cause memory leaks.
+
+5.  **IDs and Context**:
+
+    - **v2**: `trace_id` and `observation_id` were often passed around.
+    - **v3**:
+      - OTEL handles context propagation automatically. Child observations are created under the currently active span/generation in the context.
+      - To link to an existing trace from an external system, use the `trace_context={"trace_id": "...", "parent_span_id": "..."}` parameter when creating a new span/generation.
+      - Trace IDs and Observation (Span) IDs now follow W3C Trace Context format (32-char hex for trace, 16-char hex for span).
+      - Use `langfuse.create_trace_id()` for generating compliant IDs, especially for linking scores or external data.
+      - Setting custom observation IDs is not supported.
+      - `get_trace_id()` is now `get_current_trace_id()`
+
+6.  **Updating Observations**:
+
+    - **v2**: `trace.update()`, `span.update()`, `generation.update()`.
+    - **v3**:
+      - Use the `.update()` method on the `LangfuseSpan` or `LangfuseGeneration` object.
+      - To update the _currently active_ observation without a direct reference, use `langfuse.update_current_span()` or `langfuse.update_current_generation()`.
+      - For trace-level attributes, use `span_obj.update_trace()` or `langfuse.update_current_trace()`.
+      - Trace tags are only merged server-side if delivered on different OTEL spans.
+      - Trace metadata is merged server-side when delivered via different spans.
+      - Metadata for both traces and observations are merged even within same span update calls if they are dicts and on different keys (only first level).
+
+7.  **Decorator (`@observe`)**:
+
+    - **v2**: The top-most decorated function was the trace.
+    - **v3**: The top-most decorated function is now the root span. Trace updates must be done by calling `langfuse.update_current_trace()`.
```

Comment:
does trace i/o default to root span i/o or does it need to be set via the update method?